README.md

```diff
@@ -1,707 +1,1799 @@
+
+  const todosById = useSelector(selectTodosById)
+
+  return (
+    <div>
+      {todosById.map(todo => (
+        <div key={todo.id}>{todo.title}</div>
+      ))}
+    </div>
+  )
+}
+```
+
+3. Using [`useCallback`].
+
+```tsx
+const parametricSelector = createSelector(
+  [(state: RootState) => state.todos, (state: RootState, id: number) => id],
+  (todos, id) => todos.filter(todo => todo.id === id)
+)
+
+const MyComponent: FC<Props> = ({ id }) => {
+  const selectTodosById = useCallback(parametricSelector, [])
+
+  const todosById = useSelector(state => selectTodosById(state, id))
+
+  return (
+    <div>
+      {todosById.map(todo => (
+        <div key={todo.id}> {todo.title}</div>
+      ))}
+    </div>
+  )
+}
+```
+
+4. Use [`re-reselect`]:
+
+```ts
+import { createCachedSelector } from 're-reselect'
+
+const parametricSelector = createCachedSelector(
+  [(state: RootState) => state.todos, (state: RootState, id: number) => id],
+  (todos, id) => todos.filter(todo => todo.id === id)
+)((state: RootState, id: number) => id)
+```
+
+Starting in 5.0.0, you can eliminate this problem using `weakMapMemoize`.
+
+###### Using [`createSelector`]
+
+```ts
+const parametricSelector = createSelector(
+  [(state: RootState) => state.todos, (state: RootState, id: number) => id],
+  (todos, id) => todos.filter(todo => todo.id === id),
+  {
+    memoize: weakMapMemoize,
+    argsMemoize: weakMapMemoize
+  }
+)
+
+parametricSelector(state, 0) // Selector runs
+parametricSelector(state, 0)
+parametricSelector(state, 1) // Selector runs
+parametricSelector(state, 0)
+```
+
+###### Using [`createSelectorCreator`]
+
+```ts
+import { createSelectorCreator, weakMapMemoize } from 'reselect'
+
+const createSelectorWeakMap = createSelectorCreator({
+  memoize: weakMapMemoize,
+  argsMemoize: weakMapMemoize
+})
+
+const parametricSelector = createSelector(
+  [(state: RootState) => state.todos, (state: RootState, id: number) => id],
+  (todos, id) => todos.filter(todo => todo.id === id)
+)
+
+parametricSelector(state, 0) // Selector runs
+parametricSelector(state, 0)
+parametricSelector(state, 1) // Selector runs
+parametricSelector(state, 0)
+```
+
+This solves the problem of having to know and set the cache size prior to creating a memoized selector.
+
+</details>
+
+<div align="right">[ <a href="#table-of-contents">↑ Back to top ↑</a> ]</div>
+
+---
+
+<a id="autotrackmemoize"></a>
+
+#### autotrackMemoize(func) - (**`Since`** 5.0.0)
+
+<details><summary><b>Description</b></summary>
+
+Uses an "auto-tracking" approach inspired by the work of the Ember Glimmer team. It uses a Proxy to wrap arguments and track accesses to nested fields in your selector on first read. Later, when the selector is called with new arguments, it identifies which accessed fields have changed and only recalculates the result if one or more of those accessed fields have changed. This allows it to be more precise than the shallow equality checks in defaultMemoize.
```

Comment:
:hand: Add a warning under this section clarifying that it is still experimental and undergoing testing